providers/curation/github.js

```diff
@@ -169,14 +171,59 @@ class GitHubCurationService {
     return serviceGithub.repos.createFile(fileBody)
   }
 
+  // Return an array of valid patches that exist
+  // and a list of definitions that do not exist in the store
+  async _validateComponentDefinitionExists(patches) {
+    const coordinatesWithRevisions = flatten(
+      map(patches, patch =>
+        map(patch.revisions, (_, key) => {
+          return Object.assign({}, patch.coordinates, {
+            revision: key
+          })
+        })
+      )
+    )
+
+    // Check if all the definitions exists into the store
+    const validDefinitions = await this.definitionService.listAll(coordinatesWithRevisions)
+    if (coordinatesWithRevisions.length !== validDefinitions.length) {
+      const notFoundDefinitions = []
+      map(patches, patch =>
+        map(
+          patch.revisions,
+          throat(1, async (_, key) => {
+            const coordinates = Object.assign({}, patch.coordinates, {
+              revision: key
+            })
+            if (validDefinitions.findIndex(definition => isEqual(definition, coordinates)) === -1) {
+              notFoundDefinitions.push(
+                `Definition ${EntityCoordinates.fromObject(coordinates).toString()} has not been found`
+              )
+            }
+          })
+        )
+      )
+      return { isValid: false, notFoundDefinitions }
+    }
+
+    return { isValid: true, patches }
+  }
+
   async addOrUpdate(userGithub, serviceGithub, info, patch) {
     const { owner, repo, branch } = this.options
+    const { isValid, notFoundDefinitions } = await this._validateComponentDefinitionExists(patch.patches)
+    if (!isValid) {
+      notFoundDefinitions.push(
+        'The contribution has failed because some of the supplied component definitions do not exist'
+      )
+      return { errors: notFoundDefinitions }
```

Comment:
consider throwing here. from an API point of view it seems reasonable. That is, the caller passed in some non-existent coords. Like trying to open a non-existent file. 

Also, it would avoid having to return funky structured objects.

Actually I see that i suggested that at the call site a while back

Reply:
We've changed this and throwing an error as you suggested.
To test it, given this is an async method, we introduced `chai-as-promised`